### ISSUE TITLE
Module does no result-caching... slows down all web requests?

### ISSUE THREAD

**Issue by diranged** [kvspb/nginx-auth-ldap#8]
The module seems to do no result caching, and slows down all web requests significantly. Could the module cache the results a little bit, like Apache does?

**Comment by fred9176** [comment]
+1 : cache directives solved my problem and everything is working fine ;-)

**Comment by delfuego** [comment]
I know this is a closed issue, but I wanted to throw a HUGE +1 at this, and ask that it get documented in the main docs.

I was seeing a ton of re-requests for authentication and corresponding segfaults/core dumps in the nginx-auth-ldap module until I did a deep enough dive into the debug logs to see that the main driver *seemed* to be repeat attempts to connect to and re-authenticate against the LDAP server. I then realized that, unlike the Apache LDAP module, I didn't explicitly see any caching settings for nginx-auth-ldap, which led me to do a little searching and stumble upon this issue. And adding explicit caching now seems to have cleared up *all* my issues.

So... at a minimum, this should get documented; at a maximum, perhaps the cache should be enabled by default with sensible cache size and expiration time?
